Frameworks/UIKit/UITextField.mm

```diff
@@ -129,13 +129,17 @@ - (void)setText:(NSString*)text {
 
         [_secureModeLock lock];
         if (_secureTextMode) {
-            _passwordBox.password = _text;
+            if (![_passwordBox.password isEqualToString:_text]) {
+                _passwordBox.password = _text;
+            }
         } else {
-            _textBox.text = _text;
-            // Ensure caret at end of field in case we programmatically
-            // gain focus (becomeFirstResponder) after the text is set:
-            _textBox.selectionStart = [_text length];
-            _textBox.selectionLength = 0;
+            if (![_textBox.text isEqualToString:_text]) {
+                _textBox.text = _text;
+                // Ensure caret at end of field in case we programmatically
+                // gain focus (becomeFirstResponder) after the text is set:
+                _textBox.selectionStart = [_text length];
+                _textBox.selectionLength = 0;
```

Comment:
both insertion/deletion case will be guarded by this newly added check, the code right above updated the cached value when insertion/deletion happens via text/password changed listener. we so when it comes down to the newly added check, the evaluation will be true and thus no-op.  

---
In reply to: [94637079](https://github.com/Microsoft/WinObjC/pull/1604#discussion_r94637079) [](ancestors = 94637079)